src/log-uploader.ts

```diff
@@ -21,76 +18,103 @@ function getDeferred (): IDeferred {
   return { promise, resolve: res, reject: rej };
 }
 
-export class LogUploader {
-  private sendQueue: ISendLogState[] = [];
-  private hasPendingRequest = false;
-  private backoff: Backoff;
-  private environment!: string;
+export const getOrCreateLogUploader = (url: string, debugMode: boolean = false): LogUploader => {
```

Comment:
This function will ensure singletons are used for each provided URL.

Reply:
I wonder if we should not export the `LogUploader` class. I mean presumably we want people to *always* use `getOrCreateLogUploader` right, never `new LogUploader(..)`?

Reply:
Ahh nm. i got confused, this is called internally from the logger. No change needed.